Packs/IronscalesEventCollector/Integrations/IronscalesEventCollector/IronscalesEventCollector.py

```diff
@@ -249,14 +383,22 @@ def main():
                 send_events_to_xsiam(events, VENDOR, PRODUCT)
 
         elif command == "fetch-events":
+            if demisto.getLastRun().get("last_incident_time"):
+                demisto.debug("Test-IronScales: last_run buffer is not empty. using other first_fetch time.")
+                first_fetch = arg_to_datetime(demisto.getLastRun().get("last_incident_time")) or first_fetch
+                demisto.debug(f'Test-IronScales: new time = {demisto.getLastRun().get("last_incident_time")}')
             events, last_id = fetch_events_command(
                 client=client,
                 first_fetch=first_fetch,
                 max_fetch=max_fetch,
                 last_id=demisto.getLastRun().get("last_id"),
             )
+            demisto.debug("Test-IronScales: returned from fetch_event")
+            demisto.debug(f"Test-IronScales: returned data = {str(events)}, {str(last_id)}")
             send_events_to_xsiam(events, VENDOR, PRODUCT)
-            demisto.setLastRun({"last_id": last_id})
+            demisto.setLastRun(
+                {"last_id": last_id, "last_incident_time": events[-1].get("first_reported_date") if events else None}
```

Comment:
This is not correct, if we dont have new incidents, we will still use the last_incident_time we last used. Also, according the API docs you don't have first_reported_date in the response events.